app/dagger/ClockModule.java

```diff
@@ -1,14 +1,16 @@
 package dagger;
 
-import javax.inject.Singleton;
 import java.time.Clock;
 
+/**
+ * A module that provides a clock implementation.
+ */
 @Module
 public class ClockModule {
 
-    @Singleton
     @Provides
-    public Clock providesClock() {
+    public Clock clock() {
```

Comment:
You won't need a clock module:

https://github.com/playframework/playframework/blob/3bd429d9520c297a037e41edaf7f5a6f77701852/framework/src/play/src/main/java/play/components/CryptoComponents.java#L17-L20

There is a `TODO`, but you can reuse that clock anyway.